sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/AvailabilitySetDataListOfListsTests.cs

```diff
@@ -0,0 +1,239 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.ClientModel.Tests.Client;
+using System.ClientModel.Tests.Client.Models.ResourceManager.Compute;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using NUnit.Framework;
+
+namespace System.ClientModel.Tests.ModelReaderWriterTests.Models
+{
+    public class AvailabilitySetDataListOfListsTests
+    {
+        private static readonly string s_payload = File.ReadAllText(TestData.GetLocation("AvailabilitySetData/AvailabilitySetDataListOfLists.json")).TrimEnd();
+        private static readonly BinaryData s_data = new BinaryData(Encoding.UTF8.GetBytes(s_payload));
+        private static readonly List<List<AvailabilitySetData>> s_availabilitySets = ModelReaderWriter.Read<List<List<AvailabilitySetData>>>(s_data)!;
+        private static readonly string s_collapsedPayload = GetCollapsedPayload();
+
+        private static readonly HashSet<Type> s_supportedCollectionTypes =
+        [
+            typeof(List<>),
```

Comment:
Should this use `IList<>` and `IDictionary<>`?

Reply:
we need concrete constructable types not interfaces here

Reply:
the restriction no longer exists so all types works